components/infobox/wikis/deadlock/infobox_item_custom.lua

```diff
@@ -0,0 +1,220 @@
+			{name = 'Bullet Velocity', parameter = 'velocity', funct = '_positivePercentDisplay'},
+			{name = 'Bullet Lifesteal', parameter = 'bulletlifesteal', funct = '_positivePercentDisplay'},
+			{name = 'Bullet Slow Proc', parameter = 'bulletslow', funct = '_positivePercentDisplay'},
+			{name = 'Reload Time', parameter = 'reload', funct = '_positivePercentDisplay'},
+			{name = 'Bullet Shield Health', parameter = 'bulletshield'},
+			{name = 'Bullet Resist', parameter = 'bulletresist', funct = '_positivePercentDisplay'},
+			{name = 'Bullet Resist vs. NPCs', parameter = 'bulletresistnpc', funct = '_positivePercentDisplay'},
+			{name = 'Sprint Speed', parameter = 'sprintspeed'},
+			{name = 'Stamina', parameter = 'Stamina'},
+			{name = 'Slide Distance', parameter = 'slidedistance', funct = '_positivePercentDisplay'},
+			{name = 'Spirit Power', parameter = 'spiritpower'},
+			{name = 'Spirit Resist', parameter = 'spiritresist'},
+			{name = 'Spirit Shield Health', parameter = 'spiritshield'},
+			{name = 'Spirit Lifesteal', parameter = 'spiritlifesteal', funct = '_positivePercentDisplay'},
+			{name = 'Cooldown Reduction', parameter = 'cdreduction', funct = '_positivePercentDisplay'},
+			{name = 'Movement Speed', parameter = 'movespeed'},
+		}
+		widgets = caller:_getAttributeCells(attributeCells)
+		table.insert(widgets, Cell{name = 'Standard Bonus', children = {args.standardbonus}})
+		if Table.isNotEmpty(widgets) then
+			table.insert(widgets, 1, Title{children = 'Attributes'})
+		end
+		return widgets
+	elseif id == 'ability' then
+		if String.isEmpty(args.active) and String.isEmpty(args.passive) then
+			return {}
+		end
+		Array.appendWith(widgets,
+			Cell{name = 'Active', children = {args.active, args.active2, args.active3}},
+			Cell{name = 'Passive', children = {args.passive, args.passive2, args.passive3}}
+		)
+	elseif id == 'availability' then
+		if String.isEmpty(args.category) and String.isEmpty(args.tier) then return {} end
+		return {
+			Title{children = 'Type'},
+			Cell{name = 'Category', children = {caller:_categoryDisplay()}},
+			Cell{name = 'Tier', children = {args.tier}},
+		}
+	elseif id == 'recipe' then
+		if String.isEmpty(args.recipe) then return {} end
+		return {
+			Title{children = 'Components'},
+			Center{children = {args.recipe}}
+		}
+	elseif id == 'maps' then return {}
+	elseif id == 'info' then return {}
+	end
+
+	return widgets
+end
+
+---@param args table
+---@return string[]
+function CustomItem:getWikiCategories(args)
+	if not Namespace.isMain() then return {} end
+
+	local categories = {}
+	if String.isNotEmpty(args.movespeed) then
+		table.insert(categories, 'Movement Speed Items')
+	end
+
+	local possibleCategories = {
+		['Items with Active Abilities'] = 'active',
+		['Items with Passive Abilities'] = 'passive',
+	}
+	for category, requiredArg in pairs(possibleCategories) do
+		if String.isNotEmpty(args[requiredArg]) then
+			table.insert(categories, category)
+		end
+	end
+	return categories
```

Comment:
```suggestion
	local categories = Array.append({},
		String.isNotEmpty(args.movespeed) and 'Movement Speed Items' or nil,
		self:_getCategory() and (self:_getCategory() .. ' Items') or nil,
		String.isNotEmpty(args.active) and 'Items with Active Abilities' or nil,
		String.isNotEmpty(args.passive) and 'Items with Passive Abilities' or nil
	)

	return categories
```

Reply:
^ think you can do this, with your tweak of the getCategory to String.isNotEmpty @Kanopedia

Reply:
Done o7